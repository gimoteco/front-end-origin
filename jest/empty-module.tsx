import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
