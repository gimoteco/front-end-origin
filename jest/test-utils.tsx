import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
