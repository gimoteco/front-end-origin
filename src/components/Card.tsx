import * as React from 'react';
import * as Style from './Card.styles';

interface CardProps {
  title: string;
  icon: React.ReactNode;
  subtitle: string;
  children: React.ReactNode;
}

function Card({ title, icon, subtitle, children }: CardProps) {
  return (
    <Style.CardWrapper>
      <Style.CardHeaderWrapper>
        {icon}
        <Style.CardTitle>{title}</Style.CardTitle>
        <Style.CardSubtitle>{subtitle}</Style.CardSubtitle>
      </Style.CardHeaderWrapper>
      {children}
    </Style.CardWrapper>
  );
}

export default React.memo(Card);
