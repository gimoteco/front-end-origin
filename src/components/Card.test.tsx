import React from 'react';
import { render } from '../../jest/test-utils';
import Card from './Card';

describe('Card', () => {
  it('should render according snapshot', () => {
    const { container } = render(
      <Card title="Hello World" subtitle="awesome subtitle" icon={null}>
        Hello World
      </Card>
    );

    expect(container).toMatchSnapshot();
  });
});
