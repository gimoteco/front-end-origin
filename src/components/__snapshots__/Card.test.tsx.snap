// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render according snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa jnieHq"
  >
    <div
      class="sc-bxivhb iBTOsM"
    >
      <h1
        class="sc-bwzfXH tnVib"
      >
        Hello World
      </h1>
      <h2
        class="sc-htpNat gIpqKU"
      >
        awesome subtitle
      </h2>
    </div>
    Hello World
  </div>
</div>
`;
